rtos/source/TARGET_CORTEX/mbed_lib.json

```diff
@@ -81,6 +81,9 @@
         },
         "NUVOTON": {
             "idle-thread-stack-size-debug-extra": 512
+        },
+        "MCU_PSOC6_M4": {
+            "target.macros_add": ["CY_RTOS_AWARE"]
```

Comment:
@vmedcy This is the idea. Though it looks slightly less than ideal, since the macro itself is not generic Mbed. But I can't think of another way of doing this?

Reply:
@LDong-Arm - agree, this is the good way to achieve the desired level of configurability. I don't know if generic FW code provided by another vendors has similar macro switch to enable/disable RTOS awareness without depending on any specific RTOS.

Reply:
> I don't know if generic FW code provided by another vendors has similar macro switch to enable/disable RTOS awareness without depending on any specific RTOS.

Not I'm aware of. But I guess some vendors may modify a few files when importing to Mbed OS, which probably means less struggle with the build configs but more work when importing a newer version of the upstream code. I'm not exactly familiar with it.